pkg/cmd/pulumi/config.go

```diff
@@ -913,17 +913,25 @@ func looksLikeSecret(k config.Key, v string) bool {
 // getStackConfiguration loads configuration information for a given stack. If stackConfigFile is non empty,
 // it is uses instead of the default configuration file for the stack
 func getStackConfiguration(stack backend.Stack, sm secrets.Manager) (backend.StackConfiguration, error) {
+	var cfg config.Map
 	workspaceStack, err := loadProjectStack(stack)
 	if err != nil {
-		return backend.StackConfiguration{}, fmt.Errorf("loading stack configuration: %w", err)
+		// if project is unavailable, fallback to get the project's configuration from the backend
+		cfg, err = backend.GetLatestConfiguration(commandContext(), stack)
```

Comment:
This is the most worrying change for me. Currently we do not treat config stored in the backend as authoritative, if the user is missing config locally we error. With this change we'll now carry on with the config stored in the backend.

It's probably a horrible amount of re-plumbing but I wonder if we could somehow just not hit this code path for getStackConfiguration when calling "destroy"?

Reply:
I'm not really that familiar with this yet, but why isn't the config stored in the backend authoritative if there is no alternative? Is it not the last configuration that was used to deploy the stack?

I could double check if calling destroy with an empty configuration works, but would it have issues calculating things if config was blank?

Reply:
I can't recall the reasoning I asked this _years_ ago. @lukehoban or @pgavlin might have comments on if it's ok or not for us to start using backend config like this.

Reply:
I think that using this if and only if we're not under the stack's project's directory is fine.

If local configuration is available, then we treat it as authoritative for two reasons:
1. it allows the user to change configuration for the destroy (e.g. the user can replace short-lived authentication tokens)
2. w.r.t. the model, a destroy is essentially just an `update` with a null pulumi program, and the config remains an input into the `update`.